shared/src/commonMain/kotlin/org/hyperskill/app/analytic/domain/interactor/AnalyticInteractor.kt

```diff
@@ -24,17 +40,37 @@ class AnalyticInteractor(
     }
 
     suspend fun logEvent(event: AnalyticEvent, forceLogEvent: Boolean = false) {
-        analyticEngines
-            .filter { it.targetSource == event.source }
-            .forEach { it.reportEvent(event, forceLogEvent) }
+        val engines = analyticEngines.filter { it.targetSource in event.sources }
+        val userProperties = getUserProperties()
+        engines.forEach { engine -> engine.reportEvent(event, userProperties, forceLogEvent) }
         eventMonitor?.analyticDidReportEvent(event)
     }
 
     override fun setScreenOrientation(screenOrientation: ScreenOrientation) {
-        analyticEngines.forEach { it.setScreenOrientation(screenOrientation) }
+        this.screenOrientation = screenOrientation
     }
 
     override fun setAppTrackingTransparencyAuthorizationStatus(isAuthorized: Boolean) {
-        analyticEngines.forEach { it.setAppTrackingTransparencyAuthorizationStatus(isAuthorized) }
+        this.isATTPermissionGranted = isAuthorized
     }
+
+    private suspend fun getUserProperties(): AnalyticEventUserProperties =
+        coroutineScope {
+            val profileDeferred = async {
+                currentProfileStateRepository.getState(forceUpdate = false).getOrNull()
+            }
+            val subscriptionDeferred = async {
+                currentSubscriptionStateRepository.getState(forceUpdate = false).getOrNull()
+            }
```

Comment:
I hope that `SharedPreferences` and `NSUserDefaults` has an optimization mechanism in place that involves caching, to not always perform a disk read operation whenever we request a value.

Reply:
@ivan-magda 
It's easy to add an in-memory cache to the appropriate state holders.